### ISSUE TITLE
这个loss这么小正常吗

### ISSUE THREAD

**Issue by aaashinshin** [FlagOpen/FlagEmbedding#323]
作者您好，我的微调数据是22万条qa对，负例用hn_mine对所有a抽取15条构成微调数据集，我的loss如下：
{'loss': 1.2598, 'learning_rate': 0.0, 'epoch': 0.0}                                                                                                
{'loss': 0.9385, 'learning_rate': 1e-05, 'epoch': 0.0}                                                                                              
{'loss': 0.9248, 'learning_rate': 1e-05, 'epoch': 0.0}                                                                                              
{'loss': 0.4763, 'learning_rate': 9.997761862130708e-06, 'epoch': 0.0}                                                                              
{'loss': 0.481, 'learning_rate': 9.995523724261416e-06, 'epoch': 0.01}                                                                              
{'loss': 0.3123, 'learning_rate': 9.993285586392123e-06, 'epoch': 0.01}                                                                             
{'loss': 0.2258, 'learning_rate': 9.991047448522831e-06, 'epoch': 0.01}                                                                      
{'loss': 0.3416, 'learning_rate': 9.988809310653536e-06, 'epoch': 0.01}                                                                             
{'loss': 0.2485, 'learning_rate': 9.986571172784244e-06, 'epoch': 0.01}                                                                             
{'loss': 0.241, 'learning_rate': 9.984333034914951e-06, 'epoch': 0.01}                                                                              
{'loss': 0.2354, 'learning_rate': 9.98209489704566e-06, 'epoch': 0.01}
{'loss': 0.2578, 'learning_rate': 9.979856759176366e-06, 'epoch': 0.01}
{'loss': 0.1611, 'learning_rate': 9.977618621307073e-06, 'epoch': 0.01}

请问这个loss值正常吗？
我的启动脚本：
torchrun --nproc_per_node 2 \
-m FlagEmbedding.baai_general_embedding.finetune.run \
--deepspeed ds_config.json \
--gradient_checkpointing \
--output_dir output_all_ft_test1 \
--model_name_or_path bge-large-zh-v1.5 \
--train_data data/all_ft-hnm.jsonl \
--learning_rate 1e-5 \
--fp16 \
--num_train_epochs 5 \
--per_device_train_batch_size 32 \
--dataloader_drop_last True \
--normlized True \
--temperature 0.02 \
--query_max_len 128 \
--passage_max_len 512 \
--train_group_size 8 \
--negatives_cross_device \
--logging_steps 1 \
--dataloader_num_workers 12 \
--query_instruction_for_retrieval "为这个句子生成表示以用于检索相关文章："

**Comment by staoxiao** [comment]
参数看上去没有什么问题。这个loss对于简单负样本上正常的，但对于难负样本来说小了些，可能这个任务本身就不难，或者挖掘出的负样本不够难。
可以看看构造出来的数据难度是否足够，负样本太简单的话可以减小range_for_sampling的范围获取排名更难的负样本。

**Comment by aaashinshin** [comment]
好的，感谢您的宝贵意见！
